src/transformers/tokenization_utils.py

```diff
@@ -155,10 +155,12 @@ class PreTrainedTokenizer(PreTrainedTokenizerBase):
 
     def __init__(self, **kwargs):
         super().__init__(**kwargs)
-        # Added tokens
-        self.added_tokens_encoder = {}
-        self.unique_added_tokens_encoder = []
-        self.added_tokens_decoder = {}
+
+        # Added tokens - We store this for both slow and fast tokenizers
+        # until the serialization of Fast tokenizers is updated
+        self.added_tokens_encoder: Dict[str, int] = {}
+        self.added_tokens_decoder: Dict[int, str] = {}
+        self.unique_no_split_tokens: List[str] = []
```

Comment:
Some of the tokens we want to avoid splitting on are actually not added tokens but tokens already in the base vocabulary (e.g. `[MASK]` is in Albert vocab but if we don't take special care fit, it will be split by SentencePiece magic). I renamed this internal var to make this more clear.

Reply:
That's clean!